assignment/scripts/array-practice.js

```diff
@@ -59,36 +60,51 @@ console.log('The animals are now', animalArray);
 
 // 4.b. TODO: Remove the food at the end of your array & 
 //      log both the food removed and the updated array
-
-
+let removedFood = favoriteFoods.pop();
+console.log('Removed the last food', removedFood);
+console.log(favoriteFoods);
 // Example: Add an animal to the beginning using Array.unshift
 animalArray.unshift('walrus');
 console.log(`Added an animal to beginning: ${animalArray}`);
 
 // 4.c. TODO: Add a food at the beginning of the array & log the array
-
+favoriteFoods.unshift('hamburger');
+console.log('Added a favorite food to the beginning', favoriteFoods);
 
 // Example: Remove the first animal using Array.shift
 removedAnimal = animalArray.shift();
-console.log('Removed the first animal', removedAnimal);
-console.log('The animals are now', animalArray);
+console.log(`Removed the first animal: ${removedAnimal}`);
+console.log(`The animals are now ${animalArray}`);
 
 // 4.d TODO: Remove the food at the beginning of your array & 
 //     log both the food removed and the updated array
+removedFood = favoriteFoods.shift();
+console.log(removedFood, favoriteFoods);
 
 // 4.e (STRETCH) TODO: Replace the second food in your array
 //      with another one of your favorite foods.
 //      Then log the updated array.
+favoriteFoods[1] = 'wings';
+console.log(favoriteFoods);
 
 // 4.f (STRETCH) TODO: Sort your favoriteFoods array
 //     in reverse alphabetical order. Log the array.
-
+let alphabeticalFavoriteFoods = favoriteFoods.sort().reverse();
+console.log(alphabeticalFavoriteFoods);
+// Or with same array
+// console.log(favoriteFoods.sort().reverse());
+favoriteFoods.sort().reverse();
+console.log(favoriteFoods);
 // 4.g (STRETCH) TODO: Convert your array to a string
 //     putting the word "and" between each item.
 //     eg "tacos and pizza and pasta". Log the string.
+let favoriteFoodsString = favoriteFoods.toString().replaceAll(',', ' and ');
```

Comment:
You could also have used the [join](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) method here.